scapy/contrib/coap.py

```diff
@@ -150,7 +151,7 @@ def do_build(self):
         return Packet.do_build(self)
 
     def guess_payload_class(self, payload):
-        if payload[0] != b'\xff':
+        if orb(payload[0]) != 255:
```

Comment:
Why do you need this?

Reply:
Because of python 3 magic:
```
Python 3.6.0 (v3.6.0:41df79263a11, Dec 23 2016, 08:06:12) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> a = b"\xff\x01\x02\x03"
>>> a[0]
255
>>> a == b"\xff"
False
>>>
```

Reply:
Got it! `a[0] == 255` in Python 3 and `a[0] == "\xff"` in Python 2... Maybe you could use `if payload[:1] != b'\xff'`?

Reply:
Looks like a hack, but works !
Will do